thorlcr/msort/tsortl.cpp

```diff
@@ -296,8 +304,30 @@ IRowStream *ConnectMergeRead(unsigned id, IThorRowInterfaces *rowif,SocketEndpoi
 ISocketRowWriter *ConnectMergeWrite(IThorRowInterfaces *rowif,ISocket *socket,size32_t bufsize,rowcount_t &startrec,rowcount_t &numrecs)
 {
     TransferStreamHeader hdr;
-    socket->read(&hdr,sizeof(hdr));
+    unsigned remaining = sizeof(hdr);
+    byte *dst = (byte *)&hdr;
+
+    /*
+     * A client has connected at this stage, the hdr should be sent switfly.
+     * A generous 1 minute timeout between reads, if longer, timeout exception,
```

Comment:
Not sure what circumstances data is sent here - but might that 1 minute cause issues ever?

Reply:
There is a 5 minute connect timeout on the connection that writes the header,
so if there were a series of these rogue clients blocking for several minutes, getting in quickly before the Thor header write connnect, it could cause problems.
So this won't solve some DoS type scenarios.

Reply:
Is it just the header read when first connecting, that we have that timeout on, or every read of data?

I'm more concerned about breaking existing valid scenarios than not catching all hypothetical DoS attacks here...

Reply:
This is just for the header 1st send when sort connects.